cli/config_file.rs

```diff
@@ -156,6 +158,61 @@ pub const IGNORED_RUNTIME_COMPILER_OPTIONS: &[&str] = &[
   "watch",
 ];
 
+/// Filenames that Deno will recognize when discovering config.
+const CONFIG_FILE_NAMES: [&str; 2] = ["deno.json", "deno.jsonc"];
+
+pub fn discover(flags: &crate::Flags) -> Result<Option<ConfigFile>, AnyError> {
+  if let Some(config_path) = flags.config_path.as_ref() {
+    Ok(Some(ConfigFile::read(config_path)?))
+  } else {
+    let mut checked = HashSet::new();
+    for f in flags.config_path_args() {
+      if let Some(cf) = discover_inner(&f, &mut checked)? {
+        return Ok(Some(cf));
+      }
+    }
```

Comment:
This is a bit unexpected - IIRC the algorithm we agreed upon was supposed to find a config file for each of the passed arguments, not a config file that was found first. Could you clarify this @ry?

Reply:
I've been thinking about this more just now and I think this goes beyond what we've discussed in the past. I'm wondering if when running `deno fmt` or `deno lint` should we have `deno.json` files in sub directories be resepected when there's a `deno.json` in an ancestor directory.

For example, say you have:

```
sub_dir
  - deno.json
  - file.ts
deno.json
````

1. Should running `deno fmt` have the same behaviour on `file.ts` as running `deno fmt sub_dir/file.ts`?
2. Should running `deno fmt sub_dir/file.ts` have the same behaviour as running `(cd sub_dir && deno fmt file.ts)`?

I can think of a few options:

**Option 1: Resolve based on closest deno.json**

This option would mean for `deno fmt` and `deno lint` along with `deno fmt <file-path>`, it would always resolve a config file based on the closest deno.json. So includes/excludes in parent `deno.json`s would be ignored.

* Fmt/lint option resolution: Each config file resolves from the default options
* Includes/excludes: Each config file resolves from the default options

Personally, I think this is the easiest to implement and is the least complex for users to understand. It answers yes to both the questions I asked above.

**Option 2: Resolve using resolution from all deno.json files**

This would require resolving all deno.json files in all ancestor directories, then use that information to figure out the fmt/lint options and includes/excludes for a folder based on the combined result of all config files.

* Fmt/lint option resolution: Each config file resolves from the ancestor directory's options
* Includes/excludes: Each config file resolves from the ancestor directory's includes/excludes and overrides it.

This would be complex to implement. It answers yes to both the questions I asked above.

**Option 3: Use config file resolved for root of each disjointed tree**

By "disjointed tree" I mean, directory sub trees that share no common directories from the root with other directory trees. Maybe there is a better word?

This would resolve the config file for each disjointed tree. So `deno fmt` would resolve based on the cwd and use that, but `deno fmt ./child/mod.ts ../sibling/mod.ts ../sibling/other.ts` would resolve the config file for `./child` and `../sibling` to use for their respective files.

* Fmt/lint option resolution: Each config file resolves from the root of each disjointed tree
* Includes/excludes: Each config file resolves from the root of each disjointed tree

I think this is the solution we agreed upon before? It answers no to both of the questions I asked above.

Reply:
Option 1 and Option 3 are the same except applying to multiple arguments. I agree with this approach.

Or we can just find the deno.json based on the first argument and if other files have a different config, erroniously ignore it for the time being. These are corner cases that we should have tests for anyway - we can move to option 1/3 in the future.

Reply:
I believe Option 3 was exactly what we discussed previously.